contracts/token/extensions/ERC7984Freezable.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import {FHE, ebool, euint64, externalEuint64} from "@fhevm/solidity/lib/FHE.sol";
+import {FHESafeMath} from "../../utils/FHESafeMath.sol";
+import {ConfidentialFungibleToken} from "../ConfidentialFungibleToken.sol";
+
+/**
+ * Inspired by https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/186.
+ *
+ * @dev Extension of {ERC7984} that allows to implement a freezing
+ * mechanism that can be managed by an authorized account with the
+ * {_setConfidentialFrozen} function.
+ *
+ * The freezing mechanism provides the guarantee to the contract owner
+ * (e.g. a DAO or a well-configured multisig) that a specific amount
+ * of tokens held by an account won't be transferable until those
+ * tokens are unfrozen.
+ */
+abstract contract ERC7984Freezable is ConfidentialFungibleToken {
+    /// @dev Frozen amount of tokens per address.
+    mapping(address account => euint64 encryptedAmount) private _frozenBalances;
+
+    error ERC7984UnauthorizedUseOfEncryptedAmount(euint64 encryptedAmount, address user);
+
+    event Frozen(address indexed account, euint64 encryptedAmount);
```

Comment:
lets swap the order here https://docs.soliditylang.org/en/latest/style-guide.html#order-of-layout